src/Open3D/Geometry/TriangleMesh.h

```diff
@@ -738,6 +740,88 @@ class TriangleMesh : public MeshBase {
     std::vector<std::unordered_set<int>> adjacency_list_;
     /// List of uv coordinates per triangle.
     std::vector<Eigen::Vector2d> triangle_uvs_;
+
+    struct Material {
+        struct MaterialParameter {
+            union {
+                float f4[4] = {0};
+                float f3[3];
```

Comment:
just for information, when will we use `f3`, `f2`, `f`?

Reply:
Rene says that apparently they don't get used. He's planning on refactoring the whole thing in the next sprint.